docs/advanced/error_model.md

```diff
@@ -3,4 +3,7 @@ id: advanced_error_model
 title:  "Error Model"
 ---
 
-Coming soon
+## Declared Errors vs Unforeseen Defects
+A ZIO value, iow a ZIO program, has a type parameter `E` which is the type of the declared errors it can fail with. E only covers those errors, which were specified at the outset. This does mean, that the same zio program can fail in ways that couldn't be foreseen. These unforeseen situations are aka _defects_ in a zio program, and they lie outside E.
```

Comment:
```suggestion
A ZIO value has a type parameter `E` which is the type of declared errors it can fail with. `E` only covers the errors which were specified at the outset. The same ZIO value could still throw exceptions in ways that couldn't be foreseen. These unforeseen situations are called _defects_ in a ZIO program, and they lie outside E.
```